libnmstate/netapplier.py

```diff
@@ -48,5 +73,41 @@ def _apply_iface_admin_state(client, iface_state, nmdev):
         raise UnsupportedIfaceStateError(iface_state)
 
 
-class UnsupportedIfaceStateError(Exception):
-    pass
+def _index_by_name(ifaces_state):
+    return {iface['name']: iface for iface in ifaces_state}
+
+
+def _build_connection_profile(iface_desired_state):
+    CON_PROFILE_PREFIX = 'nms-'
+
+    iface_type = _translate_to_nmtype(iface_desired_state['type'])
+    settings = [
+        nm.connection.create_settings(
+            con_name=CON_PROFILE_PREFIX + iface_desired_state['name'],
+            iface_name=iface_desired_state['name'],
+            iface_type=iface_type
+        ),
+        nm.ipv4.create_settings(),
+        nm.ipv6.create_settings(),
+    ]
+    if iface_type == 'bond':
+        bond_conf = iface_desired_state['link-aggregation']
+        bond_opts = _translate_to_nmbond_options(bond_conf)
+
+        settings.append(nm.bond.create_settings(bond_opts))
+
+    return nm.connection.create_profile(settings)
+
+
+def _translate_to_nmtype(type):
```

Comment:
This could just be a dictionary:
```python
NM_TYPES = {'bond': nmclient.NM.SETTING_BOND_SETTING_NAME}
```

and you can access it with
```
NM_TYPES.get(type, 'unknown')
```

A wrapper to allow lookups in both directions will probably make sens to allow to translyte nmtypes back to nmstate types.

Reply:
Yes, good idea. I was planning to have "translators", requiring the provider to register to the map with the type and handler function. But that will fit a later stage.
For now, I can work with your suggested approach.

Reply:
Added the translator, as we need this shared between netinfo and the netapplier.